tls/s2n_ktls.h

```diff
@@ -36,7 +37,16 @@ typedef enum {
 } s2n_ktls_mode;
 
 bool s2n_ktls_is_supported_on_platform();
+S2N_RESULT s2n_ktls_get_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
 
 /* These functions will be part of the public API. */
 int s2n_connection_ktls_enable_send(struct s2n_connection *conn);
 int s2n_connection_ktls_enable_recv(struct s2n_connection *conn);
+
+/* Testing */
+/* msghdr is not `const` for sendmsg because the real implementation will need to modify
+ * ancillary data on msghdr prior to sending */
```

Comment:
Since we have a custom s2n_sendmsg_fn, configuring the ancillary data becomes an implementation detail. The default impl sets ancillary data on msghdr while the stuffer impl doesnt

https://github.com/toidiu/s2n-tls/blob/566bdf278ea4312327ca60aee348c5e99d2c53e3/tls/s2n_ktls_io.c#L223-L224

Reply:
The fewer "implementation details" in s2n_sendmsg_fn the better, because that means that you can't test those implementation details. If they can live higher in the stack, they should.